### ISSUE TITLE
RuntimeError: BindingsError: "object type not supported ..."

### ISSUE THREAD

**Issue by do-me** [pola-rs/polars#6136]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

There appears to be an issue when the apply method return a list of objects. 

Polars understand the Python objects as columns when initially reading/parsing data but when using apply there appears to be some kind of bug or I missed some parameter to define.

### Reproducible example

```python
import polars as pl

data = {"a": [1, 2], "b": [3, 4]}
df = pl.DataFrame(data)
df["a"].apply(lambda s: [{1,2},{2},{3}])

# RuntimeError: BindingsError: "object type not supported {1, 2}"
```

### Expected behavior

It should return an object type column.

### Installed versions

<details>

```
---Version info---
Polars: 0.15.14
Index type: UInt32
Platform: Linux-4.4.0-22621-Microsoft-x86_64-with-glibc2.31
Python: 3.10.8 | packaged by conda-forge | (main, Nov 22 2022, 08:26:04) [GCC 10.4.0]
---Optional dependencies---
pyarrow: 10.0.1
pandas: 1.5.2
numpy: 1.24.1
fsspec: 2022.11.0
connectorx: <not installed>
xlsx2csv: <not installed>
matplotlib: <not installed>
```

</details>

**Comment by do-me** [comment]
Thanks a lot - I am spoiled with the pandas API and its automatic backdrop.   
  
For any future readers, this line works:   
  
\`df\["a"\].apply(lambda s: \[{1,2},{2},{3}\], return\_dtype=object)\`

**Comment by ritchie46** [comment]
So you really need python sets? That will be slow.

**Comment by do-me** [comment]
That's what I also just found out. The speed is approximately the same for Pandas and Polars in this case. Probably I should shift this discussion somewhere else, but is there a way to use multiple cores for the apply method as well? 

In my case, I need to check which string elements of a set (per row in the dataframe) match a list of sets of strings and return the respective sets to the dataframe, so this was the easiest solution I found so far.
